.github/workflows/check.yml

```diff
@@ -1,44 +1,50 @@
 name: Check
 
-on:
-  push:
-    branches:
-      - main
-  pull_request:
-    branches:
-      - main
+on: [ push, pull_request ]
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
 
 jobs:
-  check:
+  rustfmt:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v6
+      with:
+        submodules: "recursive"
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: nightly
+        components: rustfmt
+    - name: Check formatting
+      run: cargo fmt --all -- --check
```

Comment:
You may want to ignore `arceos` submodule. Another solution is to remove `submodules: "recursive"` in `actions/checkout` (`rustfmt` works without reading `Cargo.toml`).